### DISCUSSION TITLE
Is it possible to get the version information about the used components?

### DISCUSSION THREAD

**Discussion by klaarseSICKAG** [synergy-design-system/synergy-design-system#425] [Q&A] [answered]
I noticed a bug in an application which uses Synergy as a foundation. I want to report it along with the version. How can I find out the exact version in the deployed application?

**Comment by schilchSICKAG** [comment, marked as answer]
Hi @klaarseSICKAG,

the components base `SynergyElement` class provides the current library version as a `static` member. You may obtain it in various ways:

## When you know there is already a synergy component on the page that you use:

In this case, just get the version via the following script:

```javascript
// Replace 'syn-button' with an element you know is available on the page
customElements.get('syn-button').version
```

## When you do not have an idea if there is a synergy component on the page:

If you do not know an element name on the page, you can also just find the first DOM node that is a synergy element. In this case, just call the following function:

```javascript
/**
 * Obtain the version of synergy
 * @param {HTMLElement} e? Optional version of the element to query [optional]
 * @returns {string} Found synergy version
 */
const synergyVersion = e => {
  const elm = e instanceof HTMLElement ? e : Array.from(document.querySelectorAll('*')).find(f => f.tagName.toLowerCase().startsWith('syn-'))
  return elm.__proto__.constructor?.version || 'No Synergy Version could be obtained'
};

// If you do now know if there is a synergy element on page, just use this:
synergyVersion(); // <- Will return the version or an error message if none was found

// If you can obtain an element, e.g. via document.getElementById
synergyVersion(document.getElementsByTagName('syn-button')[0]);
```

Hope this helps.
